### PR TITLE
refactor(storage): split JSON parsing and generation

### PR DESCRIPTION
**Pull request by coryan**
The functions to convert `g::c::s::BucketMetadata` to and from JSON
strings was getting too long and hard to grok. Soon, I will need to add
a couple more attributes, seems like a good time to split these
functions.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/7361)

### CODE REVIEW

File: google/cloud/storage/internal/bucket_metadata_parser.cc [hunk 1]
```diff
@@ -67,298 +68,444 @@ StatusOr<UniformBucketLevelAccess> ParseUniformBucketLevelAccess(
...
-      if (!v.origin.empty()) {
-        cors_as_json["origin"] = v.origin;
-      }
-      if (!v.response_header.empty()) {
-        cors_as_json["responseHeader"] = v.response_header;
-      }
-      metadata_as_json["cors"].emplace_back(std::move(cors_as_json));
+void ToJsonCors(nlohmann::json& json, BucketMetadata const& meta) {
+  if (meta.cors().empty()) return;
+
```
Comment (dbolduc) [inline google/cloud/storage/internal/bucket_metadata_parser.cc:252]: nit: most others don't have this newline. change if you want.

File: google/cloud/storage/internal/bucket_metadata_parser.cc [hunk 2]
```diff
@@ -67,298 +68,444 @@ StatusOr<UniformBucketLevelAccess> ParseUniformBucketLevelAccess(
...
+}
 
-  if (meta.has_logging()) {
-    nlohmann::json l;
-    SetIfNotEmpty(l, "logBucket", meta.logging().log_bucket);
-    SetIfNotEmpty(l, "logObjectPrefix", meta.logging().log_object_prefix);
-    metadata_as_json["logging"] = std::move(l);
-  }
+void ToJsonLogging(nlohmann::json& json, BucketMetadata const& meta) {
+  if (!meta.has_logging()) return;
```
Comment (dbolduc) [inline google/cloud/storage/internal/bucket_metadata_parser.cc:373]: nit: most others don't have this newline. change if you want.

**Review by dbolduc — APPROVED** [review]

File: google/cloud/storage/internal/bucket_metadata_parser.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/bucket_metadata_parser.cc]: Done.

File: google/cloud/storage/internal/bucket_metadata_parser.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/internal/bucket_metadata_parser.cc]: Done.

**Pull request merged**